exchanges/graphcache/src/store/data.ts

```diff
@@ -300,9 +300,10 @@ const updateRCForLink = (
     updateRCForEntity(gc, refCount, link, by);
   } else if (Array.isArray(link)) {
     for (let i = 0, l = link.length; i < l; i++) {
-      const entityKey = link[i];
-      if (entityKey) {
-        updateRCForEntity(gc, refCount, entityKey, by);
+      if (Array.isArray(link[i])) {
+        updateRCForLink(gc, refCount, link[i], by);
+      } else if (link[i]) {
+        updateRCForEntity(gc, refCount, link[i] as string, by);
       }
```

Comment:
@JoviDeCroock This is notable, because this was a potential bug. The logic here didn't allow for correct reference counting of entities in nested lists. It potentially was creating bogus keys here.